src/panels/lovelace/hui-root.ts

```diff
@@ -0,0 +1,583 @@
+import "@polymer/app-layout/app-header-layout/app-header-layout";
+import "@polymer/app-layout/app-header/app-header";
+import "@polymer/app-layout/app-scroll-effects/effects/waterfall";
+import "@polymer/app-layout/app-toolbar/app-toolbar";
+import "@polymer/app-route/app-route";
+import "@polymer/paper-icon-button/paper-icon-button";
+import "@polymer/paper-button/paper-button";
+import "@polymer/paper-item/paper-item";
+import "@polymer/paper-listbox/paper-listbox";
+import "@polymer/paper-menu-button/paper-menu-button";
+import "@polymer/paper-tabs/paper-tab";
+import "@polymer/paper-tabs/paper-tabs";
+
+import scrollToTarget from "../../common/dom/scroll-to-target";
+
+import "../../layouts/ha-app-layout";
+import "../../components/ha-start-voice-button";
+import "../../components/ha-icon";
+import { loadModule, loadCSS, loadJS } from "../../common/dom/load_resource";
+import { subscribeNotifications } from "../../data/ws-notifications";
+import { computeNotifications } from "./common/compute-notifications";
+import "./components/notifications/hui-notification-drawer";
+import "./components/notifications/hui-notifications-button";
+import "./hui-unused-entities";
+import "./hui-view";
+import debounce from "../../common/util/debounce";
+import createCardElement from "./common/create-card-element";
+import { showEditViewDialog } from "./editor/view-editor/show-edit-view-dialog";
+import { hassLocalizeLitMixin } from "../../mixins/lit-localize-mixin";
+import {
+  html,
+  LitElement,
+  PropertyDeclarations,
+  PropertyValues,
+} from "@polymer/lit-element";
+import { HomeAssistant } from "../../types";
+import { Lovelace } from "./types";
+import { LovelaceConfig } from "../../data/lovelace";
+import { HassEntities } from "home-assistant-js-websocket";
+import { navigate } from "../../common/navigate";
+import { fireEvent } from "../../common/dom/fire_event";
+import { classMap } from "lit-html/directives/classMap";
+
+// CSS and JS should only be imported once. Modules and HTML are safe.
+const CSS_CACHE = {};
+const JS_CACHE = {};
+
+class HUIRoot extends hassLocalizeLitMixin(LitElement) {
+  public narrow?: boolean;
+  public showMenu?: boolean;
+  public hass?: HomeAssistant;
+  public lovelace?: Lovelace;
+  public columns?: number;
+  public route?: { path: string; prefix: string };
+  public routeData?: { view: string };
+  private _curView: number;
+  private notificationsOpen?: boolean;
+  private _persistentNotifications?: Notification[];
+  private _haStyle?: DocumentFragment;
+
+  private _debouncedConfigChanged: () => void;
+  private _unsubNotifications?: () => void;
+
+  static get properties(): PropertyDeclarations {
+    return {
+      narrow: {},
+      showMenu: {},
+      hass: {},
+      lovelace: {},
+      columns: {},
+      route: {},
+      routeData: {},
+      _curView: {},
+      notificationsOpen: {},
+      _persistentNotifications: {},
+    };
+  }
+
+  constructor() {
```

Comment:
Types

Reply:
constructors don't have a return type, do they?

Reply:
void idk. This was more saying there are a lot of types missing